### PR TITLE
Functional tests for coivd immunization and assessments for AB#15853.

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#15853](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15853)

## Description

- Added data test id for back icon on assessment page
- Added data test id for assessment question
- Added snack bar message for Mail
- Added functional tests to test mail - address form (required inputs, cancel and submission)
- Added functional test to test print
- Added functional tests to test assessment form (required inputs, info messages for radio selection, cancel, back and submission)

Screenshot 2023-09-18 at 2 58 43 PM

Screenshot 2023-09-18 at 3 20 27 PM

## Testing

- [ ] Unit Tests Updated
- [x] Functional Tests Updated
- [ ] Not Required

## UI Changes

No but added snack bar message

Screenshot 2023-09-18 at 3 01 11 PM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Admin/Client/Pages/Covid19TreatmentAssessmentPage.razor [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 <MudLink Href="@PatientDetailsUrl">
     <MudText Typo="Typo.h5" class="mb-4">
-        <MudIcon Icon="fas fa-arrow-left" /><span class="ml-2">Back</span>
+        <MudIcon Icon="fas fa-arrow-left" data-testid="back-icon" /><span class="ml-2">Back</span>
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/Covid19TreatmentAssessmentPage.razor:16]: I'd suggest moving the data-testid to the MudLink element and renaming it `back-button`.

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js [hunk 2]
```diff
@@ -16,6 +18,292 @@ function checkAgentAuditHistory() {
...
+    ).click();
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=treatment-benefit-not-indicated]"
+    ).should("not.exist");
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=assessment-option-no]"
+    ).click();
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=treatment-benefit-not-indicated]"
+    ).should("be.visible");
```
Comment (MikeLyttle) [inline Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js:104]: Nit: could be simplified by scoping all commands to the relevant question, e.g.

```
cy.get("[data-testid=assessment-question-1]").within(() => {
    cy.get("[data-testid=assessment-option-yes]").click();
    cy.get("[data-testid=treatment-benefit-not-indicated]").should("not.exist");

    // etc.
});
```

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js [hunk 3]
```diff
@@ -16,6 +18,292 @@ function checkAgentAuditHistory() {
...
+        .contains("Required")
+        .should("be.visible");
+    cy.get("[data-testid=assessment-question-4], div")
+        .contains("Required")
+        .should("be.visible");
+
+    cy.scrollTo("bottom");
+    cy.get("[data-testid=assessment-question-5], div")
+        .contains("Required")
+        .should("be.visible");
```
Comment (MikeLyttle) [inline Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js:220]: Question 5 isn't required but 6 is.

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js [hunk 4]
```diff
@@ -46,6 +334,38 @@ describe("Patient details page as admin", () => {
         );
     });
 
+    it("Verify covid imminuization and assessment sections", () => {
```
Comment (MikeLyttle) [inline Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js:337]:
```suggestion
    it("Verify covid immunization and assessment sections", () => {
```

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js (hunk 2 above)
Comment (BrianMaki) [reply Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js]: Changed

File: Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js (hunk 3 above)
Comment (BrianMaki) [reply Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js]: Fixed.

File: Apps/Admin/Client/Pages/Covid19TreatmentAssessmentPage.razor (hunk 1 above)
Comment (BrianMaki) [reply Apps/Admin/Client/Pages/Covid19TreatmentAssessmentPage.razor]: Changed.

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Review by BronzBierd — APPROVED** [review]

**Pull request merged**